### ISSUE TITLE
vagrant-scp plugin for 2.2.18 installs 0.1.0, vagrant-scp plugin for 2.2.16 installs 0.5.7

### ISSUE THREAD

**Issue by philschwartzsonos** [hashicorp/vagrant#12504]
Hello,

Looks like there is a regression between 2.2.16 and 2.2.18.  

### Vagrant version
2.2.18

### Host operating system
Ubuntu 18.04 (though issue also occurs with the MacOS package)

### Guest operating system
Not applicable since the issue is that 2.2.18 installs vagrant-scp 0.1.0

### Vagrantfile
Not applicable since the issue is that 2.2.18 installs vagrant-scp 0.1.0

### Debug output
https://gist.github.com/philschwartzsonos/0c8b4d71ce4b5d1f929044dab9fa9fb2

### Expected behavior
2.2.18 should install vagrant-scp 0.5.7 (same version as 2.2.16)

### Actual behavior

2.2.18 installs vagrant-scp 0.1.0

### Steps to reproduce

1. 
$ vagrant --version
Vagrant 2.2.16

2. 
$ vagrant plugin list
==> vagrant: A new version of Vagrant is available: 2.2.18 (installed version: 2.2.16)!
==> vagrant: To upgrade visit: https://www.vagrantup.com/downloads.html

vagrant-scp (0.5.7, global)

3. 
install vagrant-2.2.18 package

4.
$ vagrant --version
2.2.18

5.
$ vagrant plugin list
Vagrant failed to initialize at a very early stage:

The plugins failed to initialize correctly. This may be due to manual
modifications made within the Vagrant home directory. Vagrant can
attempt to automatically correct this issue by running:

  vagrant plugin repair

If Vagrant was recently updated, this error may be due to incompatible
versions of dependencies. To fix this problem please remove and re-install
all plugins. Vagrant can attempt to do this automatically by running:

  vagrant plugin expunge --reinstall

Or you may want to try updating the installed plugins to their latest
versions:

  vagrant plugin update

Error message given during initialization: Unable to resolve dependency: user requested 'vagrant-scp (= 0.5.7)'

6.
$ vagrant plugin expunge --reinstall

This command permanently deletes all currently installed user plugins. It
should only be used when a repair command is unable to properly fix the
system.

Continue? [N]: y

All user installed plugins have been removed from this Vagrant environment!

Vagrant will now attempt to reinstall user plugins that were removed.
Installing the 'vagrant-scp' plugin. This can take a few minutes...
Fetching vagrant-scp-0.1.0.gem
Installed the plugin 'vagrant-scp (0.1.0)'!

7.
This version is outdated and does not work.  scp'ing anything from a docker provider, for example results in:

"The provider for this Vagrant-managed machine is reporting that it
is not yet ready for SSH. Depending on your provider this can carry
different meanings. Make sure your machine is created and running and
try again. Additionally, check the output of `vagrant status` to verify
that the machine is in the state that you expect. If you continue to
get this error message, please view the documentation for the provider
you're using."

8.
The output is misleading since I can ssh to the VM or container (and using the ssh-config I can use the OS scp command to copy files to/from the VM or container)

**Comment by seluniasolo** [comment]
I also ran into this issue, and here's what I've learned so far.

Vagrant 2.2.16 depends on `net-scp`, version ~> 1.2.0. [[link]](https://github.com/hashicorp/vagrant/blob/ab0e888705f7505381865f8bc6d453cbf61ff025/vagrant.gemspec#L29)

Vagrant 2.2.17+ depends on `net-scp`, version ~> 3.0.0. [[link]](https://github.com/hashicorp/vagrant/blob/6aa0e00adbbb6a2f07cb549a491fa9ef25dd88fd/vagrant.gemspec#L29)

`vagrant-scp` 0.5.7 depends on `net-scp`, version ~> 1.1. [[link]](https://github.com/invernizzi/vagrant-scp/blob/e5c7992d29649c67ca5359a7e050aa019168f7e8/vagrant-scp.gemspec#L24)

So, once you upgrade to Vagrant 2.2.17+, `vagrant-scp` 0.5.7 cannot be installed because the `net-scp` version (3.0.0) included in Vagrant 2.2.17+ no longer satisfies the dependency.

Vagrant 2.2.17+ installs `vagrant-scp` 0.1.0 because that is the latest version where all specified dependencies can be met. As it turns out, `vagrant-scp` 0.1.0 did not specify `net-scp` as a dependency at all:

- [all versions](https://rubygems.org/gems/vagrant-scp/versions) of `vagrant-scp` gem
- `vagrant-scp` [0.1.0](https://rubygems.org/gems/vagrant-scp/versions/0.1.0)
- `vagrant-scp` [0.3.0](https://rubygems.org/gems/vagrant-scp/versions/0.3.0)

### possible workaround: build updated plugin locally

Not really tested, but I was able to resume my own simple usage of `vagrant-scp` by building a slightly updated plugin locally.

1. `git clone https://github.com/invernizzi/vagrant-scp.git`

2. [OPTIONAL] Modify `lib/vagrant/scp/version.rb`. Change the value of [VERSION](https://github.com/invernizzi/vagrant-scp/blob/f526889e8b31d9e1c52a6bf58d56530731e09e97/lib/vagrant/scp/version.rb#L3) to something like `0.5.7.1`. This is mainly just to keep things more separate, and reduce the chance of confusion later on.

3. Modify `vagrant-scp.gemspec`. Update the `net-scp` [dependency](https://github.com/invernizzi/vagrant-scp/blob/f526889e8b31d9e1c52a6bf58d56530731e09e97/vagrant-scp.gemspec#L24) to something like:

  `  spec.add_runtime_dependency 'net-scp', ">= 1.1"`

(This is rather broad, but it's a local change. This is better fixed or analyzed upstream.)

4. Build a new Vagrant plugin gem: `gem build vagrant-scp.gemspec`

5. Install this updated Vagrant plugin from the gem file built in Step 4: `vagrant plugin install PATH/TO/vagrant-scp-0.5.7.1.gem`

6.
```
$  vagrant plugin list
vagrant-scp (0.5.7.1, global)
  - Version Constraint: 0.5.7.1
```

I later discovered that there was already a [similar PR](https://github.com/invernizzi/vagrant-scp/pull/40) to deal with `net-scp` 2.x. Unfortunately, it's from 2019 and it was never merged.

For what it might be worth, I tried submitting a [new PR](https://github.com/invernizzi/vagrant-scp/pull/45) with changes from above.

**Comment by chrisroberts** [comment]
Closing this as it's a plugin dependency issue and is addressed in linked PR (https://github.com/invernizzi/vagrant-scp/pull/45)
